src/Tests/Microsoft.Diagnostics.Monitoring.UnitTests/DisposableBox.cs

```diff
@@ -0,0 +1,34 @@
+﻿using System;
+using System.Threading.Tasks;
+
+namespace Microsoft.Diagnostics.Monitoring.UnitTests
+{
+    internal class DisposableBox<T> : IDisposable where T : class, IDisposable
+    {
+        public T Value { get; private set; }
+
+        public DisposableBox(T value)
+        {
+            Value = value ?? throw new ArgumentNullException(nameof(value));
+        }
+
+        public void Dispose()
+        {
+            Release()?.Dispose();
+        }
+
+        public async Task<U> GetAndReleaseAsync<U>(Func<T, Task<U>> func)
```

Comment:
Given the specific usage of this api, do we have to keep this so generic? For example, could DisposableBox be a wrapper from HttpResponse only, with a direct conversion to ApiStreamHolder? I'm just not sure we would ever use this class for lifetime management of other types.

Reply:
Sure, I'll either fold it into ApiStreamHolder/ResponseStreamHolder or some other class that is specifically for those types. But if I run into another type that I need to conditionally dispose, then I'll likely refactor it back out into its own class again.